components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Gateway/src/Server/GatewayServer.cs

```diff
@@ -1608,25 +1608,25 @@ public sealed class GatewayServer : SessionServerBase, IServer {
 
                 SecurityConfiguration = new SecurityConfiguration {
                     ApplicationCertificate = new CertificateIdentifier {
-                        StoreType = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ?
-                                "X509Store" : "Directory",
-                        StorePath = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ?
-                                "CurrentUser\\UA_MachineDefault" :
-                                "pki/own",
+                        StoreType = CertificateStoreType.Directory,
+                        StorePath = "pki/own",
                         SubjectName = "Microsoft.Azure.IIoT.Gateway"
                     },
                     TrustedPeerCertificates = new CertificateTrustList {
-                        StoreType = "Directory",
+                        StoreType = CertificateStoreType.Directory,
                         StorePath = "pki/trusted",
                     },
                     TrustedIssuerCertificates = new CertificateTrustList {
-                        StoreType = "Directory",
+                        StoreType = CertificateStoreType.Directory,
                         StorePath = "pki/issuers",
                     },
                     RejectedCertificateStore = new CertificateTrustList {
-                        StoreType = "Directory",
+                        StoreType = CertificateStoreType.Directory,
                         StorePath = "pki/rejected"
                     },
+                    MinimumCertificateKeySize = 1024,
+                    RejectSHA1SignedCertificates = true,
```

Comment:
We should allow sha1 for interop.  Let's add a workitem to warn user in the new cert validation API.

Reply:
done, DevOps backlog item created as well